errtrace.go

```diff
@@ -101,6 +102,15 @@ func FormatString(target error) string {
 	return s.String()
 }
 
+// LogAttr builds a slog attribute for an error with the key "error".
+//
+// When serialized with a slog-based logger,
+// this will report an error return trace if the error has one,
+// otherwise the original error message will be logged as-is.
+func LogAttr(err error) slog.Attr {
```

Comment:
Since the implementation of `LogAttr` is completely independent of errtrace, I'm not sure if it's valuable to add it here. I'll drop this for now, and we can always bring it back if we think there's value in this being part of the errtrace interface.